### ISSUE TITLE
Some weights of AlbertForPreTraining were not initialized from the model checkpoint at albert-base-v2 and are newly initialized: ['sop_classifier.classifier.weight', 'sop_classifier.classifier.bias']

### ISSUE THREAD

**Issue by L-Zhe** [huggingface/transformers#9319]
Albert leverage sentence coherence predict loss(SOP) to optimizer its parameters, and I wanna employ it to score the coherence between two sentences. But when I use AlbertForPreTraining to load albert-xxlarge-v2 checkpoint, it reminds me that:
_Some weights of AlbertForPreTraining were not initialized from the model checkpoint at albert-base-v2 and are newly initialized: ['sop_classifier.classifier.weight', 'sop_classifier.classifier.bias'] 
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference._ 

I try many times and find the output of each time is different, this means that the final linear layer for classification has not been loaded and just initial by random. I wanna know how to leverage pretrain SOP classification without fine-tuning?

Hope to get response rapidly.

**Comment by L-Zhe** [comment]
And I found bert has the same problem. I leverage BertForNextSentencePrediction to load bert-large-uncased checkpoint, and also meet the same warning information:

_Some weights of the model checkpoint at /home/linzhe/tool/transformers/bert-large-uncased were not used when initializing BertForNextSentencePrediction: ['cls.predictions.bias', 'cls.predictions.transform.dense.weight', 'cls.predictions.transform.dense.bias', 'cls.predictions.decoder.weight', 'cls.predictions.transform.LayerNorm.weight', 'cls.predictions.transform.LayerNorm.bias']
- This IS expected if you are initializing BertForNextSentencePrediction from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing BertForNextSentencePrediction from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model)._

Does it because all checkpoints for pretrain model have not the last classification linear layer parameter and require to fune-tunning ourselves, or something wrong with my operation? 
If I just want to use the pretrain next sentence prediction to score my dataset without fune tuning (to make sure the fair evaluation), what should I do?

**Comment by patrickvonplaten** [comment]
I'm not sure we have a completely fine-tuned SOP classification model. My best advice is to try out different models and see which model not randomly allocates the weights for those layers.
